inc/features/blocks/personalization-variant/edit.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react';
+
+const { InnerBlocks } = wp.blockEditor;
+const { compose } = wp.compose;
+const { withSelect, withDispatch } = wp.data;
+
+const Edit = ( {
+	hasChildBlocks,
+	isSelected,
+	onSelect,
+} ) => {
+	// Select the block parent if a variant is directly selected.
+	useEffect( () => {
+		if ( isSelected ) {
+			onSelect();
+		}
+	}, [ isSelected ] );
+
+	const props = {};
+	if ( ! hasChildBlocks ) {
+		// If we don't have any child blocks, show large block appender button.
+		props.renderAppender = () => <InnerBlocks.ButtonBlockAppender />;
+	}
+
+	return <InnerBlocks { ...props } />;
```

Comment:
To match Save, think this should be multi-line even those it's simple.

```suggestion
	return (
		<InnerBlocks
			{ ...props }
		/>
	);
```